src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,165 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> first;
+    private Node<T> last;
+
     @Override
     public void add(T value) {
+        linkLast(value);
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || size < index) {
+            throw new IndexOutOfBoundsException("The index " + index + " is invalid "
+                    + "the actual size of array is " + size);
+        }
+        Node<T> newNode = new Node<>(value);
+        if (first == null) {
+            first = last = newNode;
+        } else if (index == 0) {
+            newNode.next = first;
+            first = newNode;
+        } else if (index == size) {
+            last.next = newNode;
+            last = newNode;
+        } else if (index > 0 && size > index) {
+            Node<T> current = first;
+            int c = 0;
+            while (current != null && c != index) {
+                current = current.next;
+                c++;
+            }
+            current.prev.next = newNode;
+            newNode.prev = current.prev;
+            current.prev = newNode;
+            newNode.next = current;
+        }
```

Comment:
Don't iterate your nodes from the head to the tail every time. Pay attention to the index you receive as an input, and iterate only the first or second half of the list, depending on the index value. That will boost your lists' performance.